### PR TITLE
#39150 : fix incorrect borders in list-group-horizontal

### LINKED ISSUE #39150: Incorrect borders with .list-group-flush.list-group-horizontal
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

When using `.list-group-flush` and `.list-group-horizontal` on the same list group you get:

Screenshot 2023-09-05 at 15 56 33

But was expecting:

Screenshot 2023-09-05 at 15 56 33 copy

[Documentation seems to say](https://getbootstrap.com/docs/5.3/components/list-group) that this is possible:

> Add `.list-group-flush` to remove some borders and rounded corners to render list group items edge-to-edge in a parent container (e.g., cards).

---

I have seen another issue about putting a list group IN a list group. But that is not what is being reported here.

---

Workaround (how I made the second screenshot) is to add `.border-0` on each `.list-group-item`

### Reduced test cases

    
    <div class="card" style="width: 18rem;">
      <div class="card-header">
        Featured
      </div>
      <div class="list-group list-group-flush list-group-horizontal">
        <a href="#" class="list-group-item list-group-item-action">Certificates</a>
        <a href="#" class="list-group-item list-group-item-action">Products</a>
        <a href="#" class="list-group-item list-group-item-action">AHA</a>
      </div>
    </div>

    <hr>

    
    <div class="card" style="width: 18rem;">
      <div class="card-header">
        Featured
      </div>
      <div class="list-group list-group-flush list-group-horizontal">
        <a href="#" class="list-group-item list-group-item-action border-0">Certificates</a>
        <a href="#" class="list-group-item list-group-item-action border-0">Products</a>
        <a href="#" class="list-group-item list-group-item-action border-0">AHA</a>
      </div>
    </div>

### What operating system(s) are you seeing the problem on?

macOS

### What browser(s) are you seeing the problem on?

Microsoft Edge

### What version of Bootstrap are you using?

5.3.1

### PR DESCRIPTION
**Pull request by Luciferxgkp**
### Description

Closes #39150 

I Have tried to solve the issue of https://github.com/twbs/bootstrap/issues/39150 but i don't know how to test the css that i changes locally.
### Type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist

- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)
- [x] My code follows the code style of the project _(using `npm run lint`)_
- [ ] My change introduces changes to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

#### Live previews

- <https://deploy-preview-39193--twbs-bootstrap.netlify.app/>

### CODE REVIEW

**Comment by julien-deramond** [comment]
Thanks for creating this PR @Luciferxgkp 
Sorry for delay of the review.
I've checked the suggested modification but the rendering doesn't work with the use case from the [CodePen in the description](https://codepen.io/julien-deramond/pen/rNoWJxg):

```html
<!-- Follow these instructions, please! -->
<div class="container py-5">
  <!-- a card with a link list group -->
  <div class="card" style="width: 18rem;">
    <div class="card-header">
      Featured
    </div>

    <div class="list-group list-group-flush list-group-horizontal">
      <a href="#" class="list-group-item list-group-item-action">Certificates</a>
      <a href="#" class="list-group-item list-group-item-action">Products</a>
      <a href="#" class="list-group-item list-group-item-action">AHA</a>
    </div>
  </div>

  <hr class="border-0">

  <div class="list-group list-group-flush list-group-horizontal">
    <a href="#" class="list-group-item list-group-item-action">Certificates</a>
    <a href="#" class="list-group-item list-group-item-action">Products</a>
    <a href="#" class="list-group-item list-group-item-action">AHA</a>
  </div>
```

Here's the rendering with the PR:
Screenshot 2023-11-13 at 17 39 51

I will close this PR for now. Feel free to open a new one if you managed to find another solution.

> but i don't know how to test the css that i changes locally.

Basically, you have to clone the project, run `npm i` and then `npm run start`. And go for example to http://localhost:9001/docs/5.3/components/list-group/.
